Snowflake.Data/Core/FileTransfer/StorageClient/SFS3Client.cs

```diff
@@ -519,71 +519,89 @@ private GetObjectRequest GetGetObjectRequest(ref AmazonS3Client client, SFFileMe
         /// </summary>
         /// <param name="ex">Exception from file header.</param>
         /// <param name="fileMetadata">The file metadata.</param>
-        /// <returns>The file metadata.</returns>
-        private SFFileMetadata HandleFileHeaderErr(Exception ex, SFFileMetadata fileMetadata)
+        private void HandleFileHeaderErr(Exception ex, SFFileMetadata fileMetadata)
```

Comment:
for me returning fileMetadata has some sense because it emphasizes that fileMetadata parameter was modified.
But we can return void if you don't like the previous approach.

Reply:
I think it is confusing when we return SFFileMetadata and do nothing with the returned value. It seems to me that the object passed as a parameter should be immutable if we return a modified copy of it.